homeassistant/components/mysensors/device.py

```diff
@@ -178,28 +179,29 @@ async def async_update(self) -> None:
             else:
                 self._values[value_type] = value
 
-    async def _async_update_callback(self) -> None:
+    @callback
+    @abstractmethod
+    def _async_update_callback(self) -> None:
         """Update the device."""
-        raise NotImplementedError
 
     @callback
     def async_update_callback(self) -> None:
         """Update the device after delay."""
         if self._update_scheduled:
             return
 
-        async def update() -> None:
+        @callback
+        def async_update() -> None:
             """Perform update."""
             try:
-                await self._async_update_callback()
+                self._async_update_callback()
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Error updating %s", self.name)
             finally:
                 self._update_scheduled = False
 
         self._update_scheduled = True
-        delayed_update = partial(self.hass.async_create_task, update())
-        self.hass.loop.call_later(UPDATE_DELAY, delayed_update)
+        self.hass.loop.call_later(UPDATE_DELAY, async_update)
```

Comment:
Is the untracked task intentional here?

Reply:
I'll replace this with a debouncer in the next PR so I haven't bothered improving this.